libmscore/text.cpp

```diff
@@ -1811,7 +1811,7 @@ bool Text::edit(MuseScoreView*, Grip, int key, Qt::KeyboardModifiers modifiers,
 
                   case Qt::Key_Tab:
                   case Qt::Key_Space:
-                        s = " ";
+                        s = QString(QChar(0xa0)); // non-breaking space
```

Comment:
Only do this if control modifier is present. This transforms **all** spaces in normal textframes and system/staff text into NBSPs!

Reply:
Fixed, thanks!